### PR TITLE
HTTPS listner for simple axix 2 server

### PR DESCRIPTION
**Pull request by warunap**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -407,6 +407,11 @@
                 <artifactId>commons-httpclient</artifactId>
                 <version>${commons.httpclient.version}</version>
             </dependency>
+            <dependency>
+                <groupId>com.icegreen</groupId>
+                <artifactId>greenmail</artifactId>
+                <version>1.3.1b</version>
```
Comment (krishanthasamaraweera) [inline pom.xml]: version should be referred though properties

File: test-automation-framework/org.wso2.carbon.automation.extensions/src/main/java/org/wso2/carbon/automation/extensions/servers/axis2server/Axis2ServerManager.java [hunk 2]
```diff
@@ -110,6 +115,75 @@ public void deployService(Object service) throws IOException {
         cfgCtx.getAxisConfiguration().addServiceGroup(serviceGroup);
     }
 
+    /**
+     * replace key store paths of file for HTTPS transport
+     * @param file
```
Comment (krishanthasamaraweera) [inline test-automation-framework/org.wso2.carbon.automation.extensions/src/main/java/org/wso2/carbon/automation/extensions/servers/axis2server/Axis2ServerManager.java]: describe all parameters and exceptions thrown

File: test-automation-framework/org.wso2.carbon.automation.extensions/src/main/java/org/wso2/carbon/automation/extensions/servers/axis2server/Axis2ServerManager.java [hunk 3]
```diff
@@ -110,6 +115,75 @@ public void deployService(Object service) throws IOException {
...
+     * @param newFile
+     * @throws IOException
+     */
+    private void changeConfiguration(String file, String newFile) throws IOException {
+        StringBuilder sb = new StringBuilder();
+        File config =
+                new File(FrameworkPathUtil.getSystemResourceLocation() + File.separator +
+                        "artifacts" + File.separator + "AXIS2" + File.separator + "config" +
+                        File.separator + file);
+        BufferedReader br = null;
```
Comment (krishanthasamaraweera) [inline test-automation-framework/org.wso2.carbon.automation.extensions/src/main/java/org/wso2/carbon/automation/extensions/servers/axis2server/Axis2ServerManager.java]: no need to initialise values with null

File: test-automation-framework/org.wso2.carbon.automation.extensions/src/main/java/org/wso2/carbon/automation/extensions/servers/axis2server/Axis2ServerManager.java [hunk 4]
```diff
@@ -110,6 +115,75 @@ public void deployService(Object service) throws IOException {
...
+                os.close();
+
+            if (br != null)
+                br.close();
+        }
+
+    }
+
+    /**
+     * copy resources
```
Comment (krishanthasamaraweera) [inline test-automation-framework/org.wso2.carbon.automation.extensions/src/main/java/org/wso2/carbon/automation/extensions/servers/axis2server/Axis2ServerManager.java]: describe all parameters

File: test-automation-framework/org.wso2.carbon.automation.test.utils/src/main/java/org/wso2/carbon/automation/test/utils/generic/email/EmailServerUtil.java [hunk 5]
```diff
@@ -0,0 +1,112 @@
+/*
+*Copyright (c) 2005-2010, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (krishanthasamaraweera) [inline test-automation-framework/org.wso2.carbon.automation.test.utils/src/main/java/org/wso2/carbon/automation/test/utils/generic/email/EmailServerUtil.java]: Use correct license herders - year should be 2014

File: test-automation-framework/org.wso2.carbon.automation.test.utils/src/main/java/org/wso2/carbon/automation/test/utils/generic/email/EmailServerUtil.java [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+
+/**
+ * simple mail server for test cases using greenMail , http://www.icegreen.com/greenmail/readme.html
+ */
+public class EmailServerUtil {
+
+    GreenMail greenMail = null;
```
Comment (krishanthasamaraweera) [inline test-automation-framework/org.wso2.carbon.automation.test.utils/src/main/java/org/wso2/carbon/automation/test/utils/generic/email/EmailServerUtil.java]: not need of null initialization

File: test-automation-framework/org.wso2.carbon.automation.test.utils/src/main/java/org/wso2/carbon/automation/test/utils/generic/email/EmailServerUtil.java [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+     * start mail Server
+     */
+    public void startMailServer() {
+        // all protocols supported here
+        greenMail = new GreenMail();
+        greenMail.start ();
+    }
+
+    /**
+     * send simple text mail
```
Comment (krishanthasamaraweera) [inline test-automation-framework/org.wso2.carbon.automation.test.utils/src/main/java/org/wso2/carbon/automation/test/utils/generic/email/EmailServerUtil.java]: describe all parameters

File: test-automation-framework/org.wso2.carbon.automation.test.utils/src/main/java/org/wso2/carbon/automation/test/utils/generic/email/EmailServerUtil.java [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+    /**
+     * get all the received messages in the mail server, this is total messages in the server instance
+     * @return
+     */
+
+    public MimeMessage[] getReceivedMessages() {
+        return greenMail.getReceivedMessages();
+    }
+
+    public void stopMailServer () {
```
Comment (krishanthasamaraweera) [inline test-automation-framework/org.wso2.carbon.automation.test.utils/src/main/java/org/wso2/carbon/automation/test/utils/generic/email/EmailServerUtil.java]: method comment missing

File: test-automation-framework/org.wso2.carbon.automation.test.utils/src/main/java/org/wso2/carbon/automation/test/utils/generic/email/EmailServerUtil.java [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+
+    /**
+     * create user in green mail server memory
+     * @param mailAddress
+     * @param username
+     * @param password
+     * @return
+     */
+
+    public MailUser createUser (String mailAddress, String username, String password) {
```
Comment (krishanthasamaraweera) [inline test-automation-framework/org.wso2.carbon.automation.test.utils/src/main/java/org/wso2/carbon/automation/test/utils/generic/email/EmailServerUtil.java]: use char array to hold password values

File: test-automation-framework/org.wso2.carbon.automation.test.utils/src/main/java/org/wso2/carbon/automation/test/utils/generic/email/EmailServerUtil.java [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+    }
+
+    /**
+     * create mime message
+     * @param mailString
+     * @return
+     * @throws MessagingException
+     */
+
+    public MimeMessage createMailMessage(String mailString ) throws MessagingException {
```
Comment (krishanthasamaraweera) [inline test-automation-framework/org.wso2.carbon.automation.test.utils/src/main/java/org/wso2/carbon/automation/test/utils/generic/email/EmailServerUtil.java]: Recheck your styles note the space character in method parameter closing parenthesis

File: test-automation-framework/org.wso2.carbon.automation.test.utils/src/main/java/org/wso2/carbon/automation/test/utils/generic/email/MailUser.java [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+     */
+
+    public int getMessageCount() throws RuntimeException {
+
+        int mailCount = 0;
+        try {
+            mailCount = greenMail.getManagers().getUserManager().getImapHostManager().getInbox(greenMailUser).getMessageCount();
+        } catch (FolderException e) {
+            log.error("Error occurred while retrieving user mail box via greenmail", e);
+            throw new RuntimeException("Error occurred while retrieving user mail box via greenmail", e);
```
Comment (krishanthasamaraweera) [inline test-automation-framework/org.wso2.carbon.automation.test.utils/src/main/java/org/wso2/carbon/automation/test/utils/generic/email/MailUser.java]: why runtime exception - should this be a custom exception or more specific run time exception type

File: test-automation-framework/org.wso2.carbon.automation.test.utils/src/test/java/org/wso2/carbon/automation/test/utils/tests/EmailServerTestCase.java [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+ * simple test case for greenmail server
+ */
+public class EmailServerTestCase {
+
+    static final Log log = LogFactory.getLog(EmailServerTestCase.class);
+
+    EmailServerUtil emailServerUtil = null;
+
+    @BeforeClass
+    public void init() {
```
Comment (krishanthasamaraweera) [inline test-automation-framework/org.wso2.carbon.automation.test.utils/src/test/java/org/wso2/carbon/automation/test/utils/tests/EmailServerTestCase.java]: set always run ture

File: test-automation-framework/org.wso2.carbon.automation.test.utils/src/test/java/org/wso2/carbon/automation/test/utils/tests/EmailServerTestCase.java [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    @Test(groups = "email.unit.test", description = "tests basic email server functionality")
+    public void emailTransportTest()  {
+
+        for (int i = 0; i < 5; i++) {
+            emailServerUtil.sendTextEmail("to@localhost.com", "from@localhost.com", "subject", "body");
+        }
+
+        Assert.assertEquals(emailServerUtil.getReceivedMessages().length, 5, "email messages not received");
```
Comment (krishanthasamaraweera) [inline test-automation-framework/org.wso2.carbon.automation.test.utils/src/test/java/org/wso2/carbon/automation/test/utils/tests/EmailServerTestCase.java]: use static asserts

File: test-automation-framework/org.wso2.carbon.automation.test.utils/src/main/java/org/wso2/carbon/automation/test/utils/generic/email/MailUser.java (hunk 11 above)
Comment (warunap) [inline test-automation-framework/org.wso2.carbon.automation.test.utils/src/main/java/org/wso2/carbon/automation/test/utils/generic/email/MailUser.java]: i think this is a generic exception for all green mail sceanarios

File: test-automation-framework/org.wso2.carbon.automation.test.utils/src/main/java/org/wso2/carbon/automation/test/utils/generic/email/EmailServerUtil.java (hunk 10 above)
Comment (warunap) [inline test-automation-framework/org.wso2.carbon.automation.test.utils/src/main/java/org/wso2/carbon/automation/test/utils/generic/email/EmailServerUtil.java]: corrected

File: test-automation-framework/org.wso2.carbon.automation.test.utils/src/main/java/org/wso2/carbon/automation/test/utils/generic/email/EmailServerUtil.java (hunk 9 above)
Comment (warunap) [inline test-automation-framework/org.wso2.carbon.automation.test.utils/src/main/java/org/wso2/carbon/automation/test/utils/generic/email/EmailServerUtil.java]: can not pass to greenmail lib

File: test-automation-framework/org.wso2.carbon.automation.test.utils/src/main/java/org/wso2/carbon/automation/test/utils/generic/email/EmailServerUtil.java (hunk 7 above)
Comment (warunap) [inline test-automation-framework/org.wso2.carbon.automation.test.utils/src/main/java/org/wso2/carbon/automation/test/utils/generic/email/EmailServerUtil.java]: done

File: test-automation-framework/org.wso2.carbon.automation.test.utils/src/main/java/org/wso2/carbon/automation/test/utils/generic/email/EmailServerUtil.java (hunk 6 above)
Comment (warunap) [inline test-automation-framework/org.wso2.carbon.automation.test.utils/src/main/java/org/wso2/carbon/automation/test/utils/generic/email/EmailServerUtil.java]: done

File: test-automation-framework/org.wso2.carbon.automation.test.utils/src/main/java/org/wso2/carbon/automation/test/utils/generic/email/EmailServerUtil.java (hunk 5 above)
Comment (warunap) [inline test-automation-framework/org.wso2.carbon.automation.test.utils/src/main/java/org/wso2/carbon/automation/test/utils/generic/email/EmailServerUtil.java]: done

File: test-automation-framework/org.wso2.carbon.automation.extensions/src/main/java/org/wso2/carbon/automation/extensions/servers/axis2server/Axis2ServerManager.java (hunk 4 above)
Comment (warunap) [inline test-automation-framework/org.wso2.carbon.automation.extensions/src/main/java/org/wso2/carbon/automation/extensions/servers/axis2server/Axis2ServerManager.java]: done

File: test-automation-framework/org.wso2.carbon.automation.extensions/src/main/java/org/wso2/carbon/automation/extensions/servers/axis2server/Axis2ServerManager.java (hunk 3 above)
Comment (warunap) [inline test-automation-framework/org.wso2.carbon.automation.extensions/src/main/java/org/wso2/carbon/automation/extensions/servers/axis2server/Axis2ServerManager.java]: done

File: test-automation-framework/org.wso2.carbon.automation.extensions/src/main/java/org/wso2/carbon/automation/extensions/servers/axis2server/Axis2ServerManager.java (hunk 2 above)
Comment (warunap) [inline test-automation-framework/org.wso2.carbon.automation.extensions/src/main/java/org/wso2/carbon/automation/extensions/servers/axis2server/Axis2ServerManager.java]: done

File: pom.xml (hunk 1 above)
Comment (warunap) [inline pom.xml]: done

File: test-automation-framework/org.wso2.carbon.automation.test.utils/src/main/java/org/wso2/carbon/automation/test/utils/generic/email/MailUser.java [hunk 14]
```diff
@@ -0,0 +1,153 @@
...
+        return mailCount;
+    }
+
+    /**
+     * get unread message count
+     * @return
+     * @throws RuntimeException
+     */
+
+    public int getUnReadMessageCount() throws RuntimeException {
```
Comment (krishanthasamaraweera) [inline test-automation-framework/org.wso2.carbon.automation.test.utils/src/main/java/org/wso2/carbon/automation/test/utils/generic/email/MailUser.java]: Why run time exception ?

**Pull request merged**